acceptancetests/assess_upgrade.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+""" Assess upgrading juju controllers and models.
+
+Bootstrap a previous version of juju and then upgrade the controller and models
+to the new one.
+
+  - Spins up a local streams server (requires to be run on lxd or on a machine
+     with an externally routable interface.
+  - Bootstraps using the 'stable' juju then upgrades to the 'devel' version.
+  - Deploys mediawiki and mysql to ensure workloads continue working.
```

Comment:
I would maybe make this more generic to read deploys and validates representative workloads to ensure juju model is operational after upgrade